### PR TITLE
chore: remove deprecated config keys and endpoints code  2.0

### PR DESCRIPTION
**Pull request by pkdotson**
### SUMMARY
This pr removes old deprecated email job config keys and tasks. As well as remove deprecated legacy endpoint methods for sqla. 

**_deprecated api_** 
"/schemas/<int:db_id>/
"/schemas/<int:db_id>/<force_refresh>/"
"/csrf_token/", methods=["GET"]
"/dashboard/<int:dashboard_id>/published/", methods=("GET", "POST")
removes superset.task celery jobs and associated mehtods

**_deprecated config keys_**
OLD_API_CHECK_DATASET_OWNERSHIP
SCHEDULED_EMAIL_DEBUG_MODE
EMAIL_REPORTS_CRON_RESOLUTION
EMAIL_ASYNC_TIME_LIMIT_SEC
EMAIL_REPORT_BCC_ADDRESS
EMAIL_REPORTS_USER

### TESTING INSTRUCTIONS
User should not be able use methods and should not break new behavior for alerts.

### ADDITIONAL INFORMATION

- [x] Has associated issue: https://github.com/apache/superset/issues/18897
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/config.py [hunk 1]
```diff
@@ -743,7 +743,7 @@ def _try_json_readsha(filepath: str, length: int) -> Optional[str]:
 
 class CeleryConfig:  # pylint: disable=too-few-public-methods
     broker_url = "sqla+sqlite:///celerydb.sqlite"
-    imports = ("superset.sql_lab", "superset.tasks")
+    imports = "superset.sql_lab"
```
Comment (hughhhh) [inline superset/config.py:746]:
```suggestion
    imports = ("superset.sql_lab")
```

File: docker/pythonpath_dev/superset_config.py [hunk 2]
```diff
@@ -72,7 +72,7 @@ def get_env_variable(var_name: str, default: Optional[str] = None) -> str:
 
 class CeleryConfig(object):
     BROKER_URL = f"redis://{REDIS_HOST}:{REDIS_PORT}/{REDIS_CELERY_DB}"
-    CELERY_IMPORTS = ("superset.sql_lab", "superset.tasks")
+    CELERY_IMPORTS = "superset.sql_lab"
```
Comment (hughhhh) [inline docker/pythonpath_dev/superset_config.py:75]:
```suggestion   
    CELERY_IMPORTS = ("superset.sql_lab")
```

**Review by hughhhh — APPROVED** [review]
add these suggestions + fix the pre-commit test and LGTM

File: superset/config.py (hunk 1 above)
Comment (pkdotson) [reply superset/config.py:746]: precommit does not like the parenthesis  if it's just one.

Comment (eschutho) [reply superset/config.py:746]: can you do `("superset.sql_lab",)`?

**Comment by eschutho** [comment]
@pkdotson this is great! Are there any other references to deprecations in 2.0 after this PR? Other deprecations are fine, we just want to make sure that there are no other references to 2.0.

**Comment by pkdotson** [comment]
> @pkdotson this is great! Are there any other references to deprecations in 2.0 after this PR? Other deprecations are fine, we just want to make sure that there are no other references to 2.0.

Yes I believe this is the rest of the 2.0 code.

File: superset/config.py (hunk 1 above)
Comment (pkdotson) [reply superset/config.py]: yep that worked.

**Comment by eschutho** [comment]
> > @pkdotson this is great! Are there any other references to deprecations in 2.0 after this PR? Other deprecations are fine, we just want to make sure that there are no other references to 2.0.
> 
> Yes I believe this is the rest of the 2.0 code.

Super, thanks!

**Pull request merged**